src/components/swap/SwapModalFooter.tsx

```diff
@@ -41,9 +42,10 @@ const ConfirmButton = styled(ButtonError)`
   margin-top: 10px;
 `
 
-const DetailRowValue = styled(ThemedText.BodySmall)`
+const DetailRowValue = styled(ThemedText.BodySmall)<{ color?: string }>`
   text-align: right;
   overflow-wrap: break-word;
+  ${({ color }) => color && `color: ${color};`};
```

Comment:
You were passing a string into this, which could be interpolated on theme. Can you do that again, to avoid the use of a new hook (`useWarningColor`)? It'd look something like:

```
...<{ color?: keyof Theme }>`
  ...
  ${({ theme }) => color && `color: ${theme[color]}`};
```